ultravox/tools/ds_tool.py

```diff
@@ -72,10 +69,20 @@ def map_split(self, ds_split: datasets.Dataset, num_proc: int) -> datasets.Datas
         return ds_split.map(self._map_sample, num_proc=num_proc)
 
     def _map_sample(self, sample):
-        input_text = self.template.format(**sample)
+        rendered = jinja2.Template(self.template).render(**sample)
+
+        try:
+            turns = json.loads(rendered)
+            assert isinstance(turns, list)
+            assert all(isinstance(turn, dict) for turn in turns)
+            assert len(turns) > 0
+            assert turns[-1].get("role", None) == "user"
+        except json.JSONDecodeError:
+            turns = [{"role": "user", "content": rendered}]
```

Comment:
I find the `try/except` pattern is quite weird, but that was the only way I found to allow for both simple (template for the message content only) and full on control of the `messages` list.

Reply:
Not sure if that accomplishes the same thing:

1. that would require having two arguments to represent the same thing: column + template and column is only used to look this thing up in this case :?
2. You can maybe have one column, but want to add a system prompt or other stuff anyway which would not be possible in this paradigm.


Alternative is to make this an explicit choice: for example, `-j` as in JSON mode vs text mode.

Reply:
Added a `json_mode (-j)` instead to make the check explicit (i.e. no try/except needed).

Reply:
I see what you are getting at, it's the output of the template that matters here and that could be either json or str depending on the template. Using -j to make this deterministic seems like a good strategy.